invisible_cities/io/dst_io.py

```diff
@@ -24,3 +26,43 @@ def load_dst(filename, group, node):
 def load_dsts(dst_list, group, node):
     dsts = [load_dst(filename, group, node) for filename in dst_list]
     return pd.concat(dsts)
+
+class TableMismatchError(Exception):
+    def __init__(self):
+        s  = 'The table and dataframe dont match! '
+        Exception.__init__(self, s)
+
+def _store_pandas_as_tables(h5out, df, group_name, table_name, compression='ZLIB4', descriptive_string=None, str_col_length=32):
+    if len(df) == 0:
+        warnings.warn(f' dataframe is empty', UserWarning)
+    if '/'+group_name not in h5out:
+        group = h5out.create_group(h5out.root,group_name)
+    else:
+        group = getattr(h5out.root,group_name)
+    if table_name in group:
+        table=getattr(group,table_name)
+    else:
+        tabledef={}
+        for indx, colname in enumerate(df.columns):
+            coltype = df[colname].dtype.name
+            try:
+                tabledef[colname] = tb.Col.from_type(coltype, pos = indx)
+            except ValueError:
+                tabledef[colname] = tb.StringCol(32, pos = indx)
+
+        if descriptive_string is None:
+            descriptive_string = ''
+        table = make_table(h5out,
+                           group       = group_name,
+                           name        = table_name,
+                           fformat     = tabledef,
+                           description = descriptive_string,
+                           compression = compression)
```

Comment:
I think the whole `else` section should be a separate function.

Reply:
what do you mean by this? i dont think we should take out the 'make_table' function since i dont see where else would it be needed

Reply:
not that is needed elsewhere, but would be more readable. Like `else: create_the_table`. The other function could be defined with an leading underscore so it is not exported